system/core/tags.php

```diff
@@ -3,6 +3,64 @@
 	// - implement error notification if new tag exists
 	// - implement tag management page: edit, translate, remove or merge tags
 	// - implement tagcontainer pagetype
+	class HyphaTags {
+		static function findTagByLabel($lang, $label) {
+			/** @var HyphaDomElement $hyphaXml */
+			global $hyphaXml;
+			$path = 'hypha/tagList/tag/language[@id=' . xpath_encode($lang) . ' and @label=' . xpath_encode($label) . ']';
+			$node = $hyphaXml->findXPath($path)->parent()->first();
+			if (!$node)
+				return null;
+			return new HyphaTag($node);
+		}
+
+		static function findPagesWithTags(HyphaTag $tag, array $pagetypes=[], bool $includePrivate=false, $skip=0, $limit=0) {
+			/** @var HyphaDomElement $hyphaXml */
+			global $hyphaXml;
+
+			$pagefilters = '';
+			if ($pagetypes) {
+				$typefilter = function($type) { return "@type=" . xpath_encode($type); };
+				$attr_filters = array_map($typefilter, $pagetypes);
+				$pagefilters .= "[" . implode(" or ", $attr_filters) . "]";
```

Comment:
Varied non camel, underscore variable naming (as a non blocking review suggestion :-) )

Reply:
Good point. My personal preference is often lowercase underscore naming for variables, so somethings I forget using the proper style. Thanks, fixed now.